docs/credentials.md

```diff
@@ -1,250 +1,508 @@
-will prompt you to provide your MFA token code, and once you provide it the
-temporary credentials it retrieves will be valid for one hour.
+If you put these settings in a config profile other than `default`, you will
+need to specify which profile to use; see [set profile](#set-profile).
 
-    
-## Region
+The default location for the AWS config file is `~/.aws/config`. You can
+specify another location using environment variable `AWS_CONFIG_FILE`.
+
+---
+
+
+## Assume an IAM role with credentials from another profile
+
+You can assume an IAM role, using initial credentials provided by
+another profile. To do this, specify in the AWS config file the role to assume
+in `role_arn` and the name of the other profile in `source_profile`.
+
+    [profile my-profile]
+    role_arn=arn:aws:iam::123456789012:role/my-role-name
+    source_profile=my-other-profile
+
+Paws will look in both the AWS shared credentials file and the AWS config file
+for the source profile. The source profile can use any method to provide 
+credentials, e.g. fixed credentials, a credential process, etc.
+
+If you put these settings in a config profile other than `default`, you will
+need to specify which profile to use; see [set profile](#set-profile).
+
+The default location for the AWS config file is `~/.aws/config`. You can
+specify another location using environment variable `AWS_CONFIG_FILE`.
+
+---
 
-In order to use a Paws package, you must also set your AWS region.
 
-Paws will look for the region in the following four places, in order:
+## Use multifactor authentication
 
-1.  Custom service config.
-2.  `AWS_REGION` R environment variable.
-3.  `AWS_REGION` system environment variable (Mac and Linux Only).
-4.  AWS config file.
+To use multifactor authentication when assuming a role, specify in the AWS
+config file the MFA serial number in `mfa_serial`. When assuming the role,
+Paws will prompt you to enter your MFA token code.
 
-### Setting Region with Custom Service Config
+An example specifying MFA along with `source_profile` is below.
 
-Pass in the region for a given service: (requires version >= 0.1.5) 
+    [profile my-profile]
+    role_arn=arn:aws:iam::123456789012:role/my-role-name
+    source_profile=my-other-profile
+    mfa_serial=arn:aws:iam::123456789012:mfa/my-user-name
+
+With multifactor authentication, Paws will ask you to re-enter your MFA token
+code whenever it has to refresh your credentials (currently, after one hour).
+
+If you put these settings in a config profile other than `default`, you will
+need to specify which profile to use; see [set profile](#set-profile).
+
+The default location for the AWS config file is `~/.aws/config`. You can
+specify another location using environment variable `AWS_CONFIG_FILE`.
+
+---
+
+
+# Set region
+
+To use AWS, you must also set your AWS region. Paws will look for the region in
+the following places, in order:
+
+1.  In settings provided to an individual service.
```

Comment:
Similar issue with ordering as with credentials. These should also be links

Reply:
Moved to reference.